mp/src/game/shared/neo/weapons/weapon_neobasecombatweapon.cpp

```diff
@@ -617,6 +630,51 @@ bool CNEOBaseCombatWeapon::CanBePickedUpByClass(int classId)
 	return true;
 }
 
+void CNEOBaseCombatWeapon::DispatchMuzzleParticleEffect() {
+	const char* particleName = "ntr_muzzle_source";
+	const bool reselAllParticlesOnEntity = false;
+	const ParticleAttachment_t iAttachType = ParticleAttachment_t::PATTACH_POINT_FOLLOW;
+
+	int iAttachment = -1;
+	if (GetBaseAnimating()) {
+		// Find the attachment point index
+		iAttachment = GetBaseAnimating()->LookupAttachment("muzzle");
+		if (iAttachment <= 0)
+		{
+			Warning("Model '%s' doesn't have attachment '%s' to attach particle system '%s' to.\n", GetPrintName(), "muzzle", particleName);
```

Comment:
This warning seems to be triggering for the local player on client.dll. Should this be a `<0` check, instead?

Reply:
This is taken from one of the first DispatchParticleEffect functions in particle_parse.cpp so I think it should be fine
![image](https://github.com/NeotokyoRebuild/neo/assets/44210793/e4f98af5-df94-4242-a3c6-423411a32bc2)
It might be trying to attach the flash effect to a weapon that doesn't have a muzzle, should probably add a check that weapon is of type firearm or something

Reply:
I'm seeing the muzzle correctly if I silence this error with `==-1` check though, so I'm assuming there's some kind of discrepancy between client & server implementation here. Haven't looked at the details though.

Reply:
the whole reason for this function instead of calling dispatchparticleeffect is that i want to modify the filter thats created at one point so the owner of the gun being fired doesn't see their own 3rd person muzzle flash effect

Reply:
Ok, this actually only triggers if you switch to `thirdperson`. But even after returning back to `firstperson`, it keeps spamming the warning:

> ] thirdperson 
] firstperson 
Model 'TACHI' doesn't have attachment 'muzzle' to attach particle system 'ntr_muzzle_source' to.
Model 'TACHI' doesn't have attachment 'muzzle' to attach particle system 'ntr_muzzle_source' to.
Model 'TACHI' doesn't have attachment 'muzzle' to attach particle system 'ntr_muzzle_source' to.
Model 'TACHI' doesn't have attachment 'muzzle' to attach particle system 'ntr_muzzle_source' to.
Model 'TACHI' doesn't have attachment 'muzzle' to attach particle system 'ntr_muzzle_source' to.
Model 'TACHI' doesn't have attachment 'muzzle' to attach particle system 'ntr_muzzle_source' to.
Model 'TACHI' doesn't have attachment 'muzzle' to attach particle system 'ntr_muzzle_source' to.

Reply:
If I remember correctly you can also no longer see any muzzle flashes at all when back in first person?

Reply:
> If I remember correctly you can also no longer see any muzzle flashes at all when back in first person?

I can still see a bot's muzzle fine after the `thirdperson` swap. I think the error is only happening because the local client is attempting to draw their own muzzle, and failing (even in `firstperson`, after you've entered `thirdperson` previously).

Reply:
I think I resolved the issue, muzzle flashes in thirdperson are disabled for the owner, need to figure out a way to check if a player is in thirdperson mode or not to re-enable them